### PR TITLE
Add user-agent to credential pipelines

### LINKED ISSUE #9014: Send user-agent to AAD

### PR DESCRIPTION
**Pull request by chlowell**
Closes #9014 by ensuring credentials' requests include a user-agent, e.g. 'azsdk-python-identity/1.1.1 Python/3.7.2 (Windows-10-10.0.18362-SP0)'.

### CODE REVIEW

**Comment by chlowell** [comment]
@scbedd apparently the tests added here are failing, as if they're running against old code. They pass locally (Linux, tox). The built wheel contains the new code. I'm stumped.

File: sdk/identity/azure-identity/tests/test_authn_client.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
 from azure.core.credentials import AccessToken
 from azure.identity._authn_client import AuthnClient
 from six.moves.urllib_parse import urlparse
-from helpers import mock_response
+
+from helpers import mock_response, Request
```
Comment (bryevdv) [inline sdk/identity/azure-identity/tests/test_authn_client.py]: I looked through the file, not seeing where `Request` is used, am I missing it?

**Comment by scbedd** [comment]
@chlowell I'm checking.

File: sdk/identity/azure-identity/tests/test_authn_client.py (hunk 1 above)
Comment (chlowell) [reply sdk/identity/azure-identity/tests/test_authn_client.py]: No, that's an unused import, thanks for spotting it.

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Pull request merged**